runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/state/InMemoryBagUserStateFactory.java

```diff
@@ -80,29 +82,33 @@ public void resetForNewKey() {
     }
 
     @Override
-    public Iterable<V> get(K key, W window) {
+    public BagWithCacheToken<V> get(K key, W window) {
       initStateInternals(key);
       StateNamespace namespace = StateNamespaces.window(windowCoder, window);
       BagState<V> bagState = stateInternals.state(namespace, stateTag);
-      return bagState.read();
+      return new BagWithCacheToken<>(bagState.read(), null);
     }
 
     @Override
-    public void append(K key, W window, Iterator<V> values) {
+    @Nullable
+    public ByteString append(K key, W window, Iterator<V> values) {
       initStateInternals(key);
       StateNamespace namespace = StateNamespaces.window(windowCoder, window);
       BagState<V> bagState = stateInternals.state(namespace, stateTag);
       while (values.hasNext()) {
         bagState.add(values.next());
       }
+      return null;
```

Comment:
just for consistency I would have similar cache token implementation for this class as well.

Reply:
+1